File: solutions/julia/name-badges/1/name-badges.jl
function print_name_badge(id, name, department)

    (ismissing(id) ? "" : "[$id] - ") * name * 
    (department === nothing ? " - OWNER" : " - $(uppercase(department))")

end

function salaries_no_id(ids, salaries)
    
    sum(salary for (id, salary) in zip(ids, salaries) if ismissing(id), init=0)

end



 

File: solutions/r/name-badges/1/name-badges.R
library(stringr)

print_name_badge <- function(id, name, department) {
  if (is.na(id) && is.null(department)){
    str_glue("{name} - OWNER")
  } else if (is.na(id)){
    str_glue("{name} - {str_to_upper(department)}")
  } else if (is.null(department)) {
    str_glue("[{id}] - {name} - OWNER")
  } else {
    str_glue("[{id}] - {name} - {str_to_upper(department)}")
  }
}

salaries_no_id <- function(ids, salaries) {
  sum(salaries[is.na(ids)])
}
